apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/shard/helper/BlockDeleteHelper.java

```diff
@@ -48,13 +49,13 @@ public long processOperation(Connection sourceConnect, Connection targetConnect,
         PaginateResultWrapper paginateResultWrapper = new PaginateResultWrapper();
         paginateResultWrapper.lowerBoundColumnValue = lowerBoundIdValue;
         paginateResultWrapper.upperBoundColumnValue = upperBoundIdValue;
-
+        Set<Long> excludeDbIds = operationParams.excludeInfo != null ? operationParams.excludeInfo.getNotDeleteDbIds(): Set.of();
         try (PreparedStatement ps = sourceConnect.prepareStatement(sqlToExecuteWithPaging)) {
             do {
                 ps.setLong(1, paginateResultWrapper.lowerBoundColumnValue);
                 ps.setLong(2, paginateResultWrapper.upperBoundColumnValue);
                 ps.setLong(3, operationParams.batchCommitSize);
-            } while (handleResultSet(ps, paginateResultWrapper, sourceConnect, operationParams));
+            } while (handleResultSet(ps, paginateResultWrapper, sourceConnect, operationParams, excludeDbIds));
```

Comment:
It seems like there is no need to pass 'excludeDbIds' as separate params, because that data already passed into inside 'operationParams'. So you can get that Set